src/main/java/Duke.java

```diff
@@ -1,13 +1,47 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.regex.Pattern;
 
 public class Duke {
-    public static void main(String[] args) throws InvalidCommandException,EmptyDescriptionException {
+    public static void main(String[] args) throws InvalidCommandException,EmptyDescriptionException, IOException {
         System.out.println("Hello! I'm Duke\nWhat can I do for you?");
+
         Scanner scanner = new Scanner(System.in);
         List<Task> list = new ArrayList<>();
         Task task;
+        FileWriter fileWriter=null;
+
+        try {
+            File file = new File("./data/duke.txt");
```

Comment:
Consider declaring and using `final String` for hard-coded Strings for ease of future modifications (if necessary).